src/adapters/index.ts

```diff
@@ -0,0 +1,14 @@
+import { SupabaseClient } from "@supabase/supabase-js";
+import { Context } from "../types/context";
+import { User } from "./supabase/helpers/user";
+import { Super } from "./supabase/helpers/supabase";
+import { Database } from "./supabase/types/database";
+
+export function createAdapters(supabaseClient: SupabaseClient<Database>, context: Context) {
+  return {
+    supabase: {
+      user: new User(supabaseClient, context),
+      super: new Super(supabaseClient, context),
+    },
+  };
+}
```

Comment:
I think we should use each organization's own data stores. 

In exchange for having a universal/system wide registration system, we receive the following benefits:

1. decentralized data storage (within each org)
2. each org can edit the values as they see fit, providing them full control over their org and bot behaviors within. 
3. not needing to manage/maintain a supabase database (which can adversely affect the services of all of our users)
4. easier maintenance on the plugins, such as this, for not having to spin up a supabase.

Reply:
I've branch from the latest commit of this PR and threw something together quickly but I have a question

I'm assuming there would be a dedicated `ubiquibot-plugin-storage` repo the partner would create which would be a centralized place which plugins for that install could use?

The incoming payload has the repo in which the `/start` command was used so I can't read/write to every repo and create a `start-stop-storage.json` but I could from a dedicated repo easily. 

The impl has pulled gh ids and wallets from our sb and has written to `Record<id, wallet>` then all sb logic removed and replaced with read/write/commit to repo but without the above being true, where exactly would each plugin store the data in the partner's org?

<hr/>

I think it's a good idea because any installed plugins by a partner should be trusted so each plugin having the ability to easily access the storage of other installed plugins makes things really easy for plugin DX and interoperability.

Even if the storage repo is private just knowing the storage structure of various plugins makes it easier to build on top of things. E.g. you could name the storage for this simply `wallets.json` and then other plugins need to confirm a registered user for that org checks `wallets.json`, further, `/wallet` could read/write to the same file separating logic but sharing storage

Reply:
> I've branched from the latest commit of this PR and threw something together quickly but I have a question
> 
> 
> 
> I'm assuming there would be a dedicated `ubiquibot-plugin-storage` repo the partner would create which would be a centralized place which plugins for that install could use?
> 

`.ubiquibot-config` repo 

> 
> The incoming payload has the repo in which the `/start` command was used so I can't read/write to every repo and create a `start-stop-storage.json` but I could from a dedicated repo easily. 
> 

Sorry but this doesn't make sense to me. Can you elaborate on why it's impossible to write to the event source repository?

> 
> The impl has pulled gh ids and wallets from our sb and has written to `Record<id, wallet>` then all sb logic removed and replaced with read/write/commit to repo but without the above being true, where exactly would each plugin store the data in the partner's org?
> 

Please never use acronyms in our technical discussions. 

1. It increases cognitive overhead 
2. It reduces clarity
3. It wastes time because now I need to ask you what is "impl" before I can provide you a useful response. 
4. Other contributors with less context are less likely to be able to contribute to the conversation. 

I would only accept the use of widely used acronyms that people understand without onboarding to our project. 

> 
> <hr/>
> 
> 
> 
> I think it's a good idea because any installed plugins by a partner should be trusted so each plugin having the ability to easily access the storage of other installed plugins makes things really easy for plugin DX and interoperability.
> 
> 
> 
> Even if the storage repo is private just knowing the storage structure of various plugins makes it easier to build on top of things. E.g. you could name the storage for this simply `wallets.json` and then other plugins need to confirm a registered user for that org checks `wallets.json`, further, `/wallet` could read/write to the same file separating logic but sharing storage 
> 
> 
> 
> <hr/>
> 
> 
> 
> I forget the term you used before but essentially the pre-packaged plugins we'd advise partners to install. You'd have the bare essentials like `/start|/stop`, `/wallet` etc so with that they'd each define a spot in the storage repo which could have rules like you mentioned so it can be modified or removed but it cannot be renamed and must adhere to a specific data structure, otherwise, go nuts